crates/pessimistic-proof/src/local_state.rs

```diff
@@ -17,34 +18,50 @@ pub struct LocalNetworkState {
     pub exit_tree: LocalExitTree<Keccak256Hasher>,
     /// Commitment to the balance for each token.
     pub balance_tree: BalanceTree,
+    /// Commitment to the Nullifier Set for the local network, tracks claimed assets on foreign networks
+    pub nullifier_set: NullifierSet,
 }
 
 impl LocalNetworkState {
-    /// Apply the [`Certificate`] on the current [`State`].
+    /// Apply the [`BatchHeader`] on the current [`State`].
```

Comment:
Totally fine to rename it to something else. Initially this structure was referred as `Batch` here as well. We moved away from the `Batch` naming as it conflicts with the `Batch` on the CDK side.

What the CDK submits are actually aggregation of `Batch`. Therefore we renamed it to something else.
Either we force the CDKs to submit proofs for each and every batch individually, or for consistency we cannot use the term `BatchHeader`.

Reply:
Hmm - understood, I think that maybe it's fine to say that a set of batches is still a batch within the context of the PP. 

When you mention forcing a CDK to submit proofs for each batch individually, do you mean the PP or the validity proof?

I get that there's some specific metadata that's used to denote a batch in the CDK implementation, but I **think** that it's fine to treat some span of batches as a single batch from the perspective of the PP. This means that from the perspective of the PP,  the batch_header for a set of batches submitted by a CDK chain would be the union of all `imported_lers`, `imported_bridge_exits` and `bridge_exits`.

Reply:
Could also rename to something like BatchSetHeader if we're worried specifically about collisions with the CDK implementation.

Reply:
>When you mention forcing a CDK to submit proofs for each batch individually, do you mean the PP or the validity proof?

Ah I meant for whichever update, but in the current stage specifically for the PP.

>Could also rename to something like BatchSetHeader if we're worried specifically about collisions with the CDK implementation.

Hm but then that's not really an header as it contains the bridge exits among other payload not present in batch headers, we specifically moved away from the term `Batch` as in practice it was bringing confusions in the discussions, often we had to specify as you did by saying in the "agglayer perspective" or in the "pp perspective", ideally we want a term that does not refer already to something different to avoid having to precise the context for each occurrence.

Totally ok with moving away from `Certificate` but I do not have better naming alternative in mind right now unfortunately